### ISSUE TITLE
Invalid quickfix output for code fix all for `infer type from usage`

### ISSUE THREAD

**Issue by mhegazy** [Microsoft/TypeScript#24148]
**TypeScript Version:**  2.9.0-dev.201xxxxx

**Search Terms:** 

**Code**

```ts
let x = null;

function foo () {
    if (!x) {  // CodeFix all
        x = 22;
    }
}

function bar() {
    console.log(x);
}
```

**Expected behavior:**
```ts
let x: number | null = null;

function foo () {
    if (!x) {
        x = 22;
    }
}


function bar() {
    console.log(x);
}
```
**Actual behavior:**
```ts
let x: number: number: number = null;

function foo () {
    if (!x) {
        x = 22;
    }
}

function bar() {
    console.log(x);
}
```

**Playground Link:**  

**Related Issues:**

**Comment by andy-ms** [comment]
Fixed the duplicate error in #24169; moved the missing `null` issue to #24168 since it's a separate problem.

**Comment by andy-ms** [comment]
Never mind, that one's easy to fix too so did both in #24169.
